### PR TITLE
feat(scan): support offline option

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This option allows avoiding Internet access. The results with/without the option may differ. For example, the offline mode will not try to resolve transitive dependencies in pom.xml when the dependency doesn't exist in the local repositories. On the other hand, the online mode will resolve those dependencies in the remote repositories. It means the dependencies might be less in offline mode.

## Supported
- pom.xml
- JAR

## Issues
Close https://github.com/aquasecurity/trivy/issues/1294
Close https://github.com/aquasecurity/trivy/issues/1290
Close https://github.com/aquasecurity/trivy/issues/1173
Close https://github.com/aquasecurity/trivy/issues/1057
Close https://github.com/aquasecurity/trivy/issues/982

## Blockers
- [x] https://github.com/aquasecurity/fanal/pull/349
- [x] https://github.com/aquasecurity/go-dep-parser/pull/18
- [x] https://github.com/aquasecurity/go-dep-parser/pull/70

### CODE REVIEW

**Review by AndreyLevchenko — CHANGES_REQUESTED** [review]

File: docs/getting-started/cli/client.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ OPTIONS:
    --timeout value             timeout (default: 5m0s) [$TRIVY_TIMEOUT]
    --ignore-policy value       specify the Rego file to evaluate each vulnerability [$TRIVY_IGNORE_POLICY]
    --list-all-pkgs             enabling the option will output all packages regardless of vulnerability (default: false) [$TRIVY_LIST_ALL_PKGS]
+   --offline                   try to scan without Internet access (default: false) [$TRIVY_OFFLINE]
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/getting-started/cli/client.md:25]: Any suggestion?

[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/getting-started/cli/client.md:25]: `--offline-scan` is better?

[the remark this answers is not in the document]
Comment (AndreyLevchenko) [reply docs/getting-started/cli/client.md:25]: Something like this:
"Do not download extra information about dependencies . Some dependency parsers (like Maven) uses public API to download entire dependency tree. Use this option to disable that behavior"

[the remark this answers is not in the document]
Comment (AndreyLevchenko) [reply docs/getting-started/cli/client.md:25]: `--parse-offline`?

[the remark this answers is not in the document]
Comment (AndreyLevchenko) [reply docs/getting-started/cli/client.md:25]: or `--parse-deps-offline`

[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/getting-started/cli/client.md]: Thanks. It is a bit long for a help message. I fixed it based on your suggestion. I think it is clear enough to understand this option doesn't affect DB download.
https://github.com/aquasecurity/trivy/pull/1511/commits/558b8b72ea77d0236dcb811cdc0c6125386d4ae5

File: docs/advanced/air-gap.md [hunk 2]
```diff
@@ -50,11 +50,11 @@ $ rm trivy-offline.db.tgz
 
 In an air-gapped environment it is your responsibility to update the Trivy database on a regular basis, so that the scanner can detect recently-identified vulnerabilities. 
 
-### Run Trivy with --skip-update option
-In an air-gapped environment, specify `--skip-update` so that Trivy doesn't attempt to download the latest database file.
+### Run Trivy with --skip-update and --offline option
+In an air-gapped environment, specify `--skip-update` and `--offline` so that Trivy doesn't attempt to download the latest database file and issue any API requests for vulnerability scanning.
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/advanced/air-gap.md]: Done https://github.com/aquasecurity/trivy/pull/1511/commits/558b8b72ea77d0236dcb811cdc0c6125386d4ae5

**Review by AndreyLevchenko — APPROVED** [review]

**Pull request merged**